android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt

```diff
@@ -124,6 +133,50 @@ class JsonConverter {
             return tweaksConfig
         }
 
+        /**
+         * Converts any JS object into an `AdvertisingConfig` object.
+         * @param json JS object representing the `AdvertisingConfig`.
+         * @return The generated `AdvertisingConfig` if successful, `null` otherwise.
+         */
+        @JvmStatic
+        fun toAdvertisingConfig(json: ReadableMap?): AdvertisingConfig? = json?.getArray("schedule")
+            ?.toList<ReadableMap>()
+            ?.mapNotNull(::toAdItem)
+            ?.let { AdvertisingConfig(it) }
+
+        /**
+         * Converts any JS object into an `AdItem` object.
+         * @param json JS object representing the `AdItem`.
+         * @return The generated `AdItem` if successful, `null` otherwise.
+         */
+        @JvmStatic
+        fun toAdItem(json: ReadableMap?): AdItem? = json?.getArray("sources")
+            ?.toList<ReadableMap>()
+            ?.mapNotNull(::toAdSource)
+            ?.let { AdItem(it.toTypedArray(), json.getString("position") ?: "pre") }
```

Comment:
I think in this case it makes sense to make the code not so dense. It might be too confusing with using the `json` then again in the `let` block.

What do you think about something like this? 🙂 
```suggestion
        fun toAdItem(json: ReadableMap?): AdItem? {
            val sources = json?.getArray("sources")
                ?.toList<ReadableMap>()
                ?.mapNotNull(::toAdSource)
                ?.toTypedArray()
                ?: return null

            return AdItem(sources, json.getString("position") ?: "pre")
        }
```

Reply:
Sure, this looks better to read indeed. I'll change it.

Reply:
Done 209d875059ba7be67da4343e870fe4b18511b542